public/app/features/dashboard/components/ShareModal/ShareImage.tsx

```diff
@@ -0,0 +1,174 @@
+
+enum ImageFormat {
+  PNG = 'png',
+  JPG = 'jpg',
+}
+
+const styles = {
+  loadingContainer: {
+    margin: '16px 0',
+    textAlign: 'center' as const,
+  },
+  loadingText: {
+    marginTop: '16px',
+    color: 'var(--text-secondary)',
+  },
+  imageContainer: {
+    margin: '16px 0',
+    textAlign: 'center' as const,
+  },
+  image: {
+    maxWidth: '100%',
+    maxHeight: '400px',
+    objectFit: 'contain' as const,
+  },
+};
+
+export function ShareImage({ dashboard, panel, onDismiss }: Props) {
+  const [format, setFormat] = useState<ImageFormat>(ImageFormat.PNG);
+  const [isLoading, setIsLoading] = useState(false);
+  const [imageBlob, setImageBlob] = useState<Blob | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  const onFormatChange = (newFormat: ImageFormat) => {
+    setFormat(newFormat);
+  };
+
+  const onExport = async () => {
+    setIsLoading(true);
+    setError(null);
+
+    try {
+      if (!config.rendererAvailable) {
+        throw new Error('Image renderer service is not available');
+      }
+
+      // Use the appropriate URL building function based on whether we're sharing a dashboard or panel
+      const imageUrl = panel
+        ? buildImageUrl(true, dashboard.uid, config.theme2.isDark ? 'dark' : 'light', panel)
+        : buildDashboardImageUrl(true, dashboard.uid, config.theme2.isDark ? 'dark' : 'light', dashboard);
+
+      // Fetch the image as a blob
+      const response = await lastValueFrom(
+        getBackendSrv().fetch<BlobPart>({
+          url: imageUrl,
+          responseType: 'blob',
+        })
+      );
+
+      if (!response.ok) {
+        throw new Error(`Failed to generate image: ${response.status} ${response.statusText}`);
+      }
+
+      const blob = new Blob([response.data], { type: `image/${format}` });
+      setImageBlob(blob);
+
+      DashboardInteractions.toolbarShareClick();
+    } catch (error) {
+      console.error('Error exporting image:', error);
+      setError(error instanceof Error ? error.message : 'Failed to generate image');
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const onSave = () => {
+    if (!imageBlob) {
+      return;
+    }
+
+    const time = new Date().getTime();
+    const name = panel ? panel.title : dashboard.title;
+    saveAs(imageBlob, `${name}-${time}.${format}`);
+  };
+
+  const formatOptions: Array<SelectableValue<ImageFormat>> = [
+    { label: t('share-modal.image.format-png', 'PNG'), value: ImageFormat.PNG },
+    { label: t('share-modal.image.format-jpg', 'JPG'), value: ImageFormat.JPG },
+  ];
+
+  return (
+    <>
+      <p>
+        <Trans i18nKey="share-modal.image.info-text">
+          Export the {{ type: panel ? 'panel' : 'dashboard' }} as an image file. The image will be captured at high
+          resolution.
+        </Trans>
+      </p>
+      {/* TODO: Replace this with general message from other areas */}
+      {!config.rendererAvailable && (
```

Comment:
alliteratively we can just avoid showing the image section if the renderer is not available - depends on if we want to increase adoption of OSS for image renderer usage or something :s